api/api-public-tenant-service-account.go

```diff
@@ -117,6 +120,134 @@ func (s *server) ListServiceAccounts(ctx context.Context, in *pb.ListServiceAcco
 	}, nil
 }
 
+func (s *server) UpdateServiceAccount(ctx context.Context, in *pb.UpdateServiceAccountRequest) (res *pb.InfoServiceAccountResponse, err error) {
+	idRequest := in.GetId()
+	nameRequest := in.GetName()
+	enabledRequest := in.GetEnabled()
+	permisionsIDs := in.GetPermissionIds()
+	if idRequest == "" {
+		return nil, status.New(codes.InvalidArgument, "an id is needed").Err()
+	}
+	if nameRequest == "" {
+		return nil, status.New(codes.InvalidArgument, "an name is needed").Err()
+	}
+	if len(permisionsIDs) == 0 {
+		return nil, status.New(codes.InvalidArgument, "a list of permissions is needed").Err()
+	}
+	// start app context
```

Comment:
this should error out immediately if they try to update the access_key

Reply:
access_key is not defined on the proto as a Request so we don't care if they send it.